src/main/java/tatracker/logic/parser/SortCommandParser.java

```diff
@@ -0,0 +1,46 @@
+package tatracker.logic.parser;
+
+import static tatracker.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static tatracker.logic.parser.CliSyntax.PREFIX_GROUP;
+import static tatracker.logic.parser.CliSyntax.PREFIX_MODULE;
+import static tatracker.logic.parser.CliSyntax.PREFIX_TYPE;
+
+import tatracker.logic.commands.SortCommand;
+import tatracker.logic.commands.SortGroupCommand;
+import tatracker.logic.commands.SortModuleCommand;
+import tatracker.logic.parser.exceptions.ParseException;
+
+/**
+ * Parses input arguments and creates a new SortCommand object
+ */
+public class SortCommandParser implements Parser<SortCommand> {
+    /**
+     * Parses the given {@code String} of arguments in the context of the SortCommand
+     * and returns a SortCommand object for execution.
+     *
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public SortCommand parse(String args) throws ParseException {
+        ArgumentMultimap argMultimap = ArgumentTokenizer.tokenize(args, PREFIX_TYPE,
+                PREFIX_MODULE, PREFIX_GROUP);
+
+        if (!argMultimap.getValue(PREFIX_TYPE).isPresent()) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, SortCommand.MESSAGE_USAGE));
+        }
+
+        String type = ParserUtil.parseValue(argMultimap.getValue(PREFIX_TYPE).get());
```

Comment:
Feels like type should be parsed into an enum instead, like a SortType enum.
Cos in your commands you are doing something like a switch-case statement, but with if statements and equalsIgnoreCase.

Picked this up because it felt like type could be parsed and trimmed before it is used to create a command. That way you don't have to use equalsIgnoreCase at every step.

Reply:
Also actually parseValue is not needed if you only need to trim the type value!
argMultimap.getValue(PREFIX_TYPE).map(String::trim).get()